torch/csrc/distributed/autograd/context/thread_local_context.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <torch/csrc/distributed/autograd/context/context.h>
+
+namespace torch {
+namespace distributed {
+namespace autograd {
+
+class ThreadLocalDistAutogradContext {
+ public:
+  ThreadLocalDistAutogradContext();
+  explicit ThreadLocalDistAutogradContext(ContextPtr::weak_type&& context_wp);
+  ~ThreadLocalDistAutogradContext();
+
+  ContextPtr::weak_type context_weak_ptr_;
```

Comment:
This shouldn't be public and ideally we should expose the thread local value via a static method.